### DISCUSSION TITLE
Next features in Projectify?

### DISCUSSION THREAD

**Discussion by NicolasPetton** [NicolasPetton/Projectify#15] [Q&A]
Hi!

I have now implemented all the features I really care about in, so I was
wondering what other people might miss in the plugin.

On my todo-list I have the following (in no particular order):

* [ ] Add support for recurring tasks
* [ ] Add support for project templates (to be used when creating new
  projects)
* [ ] Add documentation about implementing specific workflows (GTD setup, etc.)
* [ ] Add a kanban view
* [x] ~Add a search input for filtering the project list/cards~
* [x] ~Add support for filtering by tags in the project list/cards~
* [x] ~Make project card colours customizable~ 

Is there anything else you think should be added to the plugin? Anything
from the list above that should be prioritized?

Thanks!
Nico

**Comment by arunnbabu81** [comment]
These were the suggestions which i could think of

1. Once a todo is completed, show the actual due date and eventual completed date to the right or below the completed todo. This gives us an estimate of time taken to finish a particular todo - useful especially in cases of overdue items.
2. Give an option to assign priority to the todo and show high priority todos as a separate heading in the scheduler.
3. Provide a peek of the todos within each project card from the dashboard itself. Similar to the icon for color change at the right top of the project card, add an icon in the right bottom of the project card. On pressing this icon, the project card will enlarge and show the todos within. The option to change the color of the card can be given within this enlarged card itself in the right top. This avoids the need to open the project card as a separate tiddler if you just want to peek into the todos of one particular project card. Not sure whether its implementable.
4. An easy way to navigate from the project page back into the dashboard by providing a button in the projects page.
5. Give an option to assign a specific time (like 10 am) or specific period (morning or evening or night) of the due date selected for that todos.

All the items in your todo-list and suggestions from the google group are also welcome additions.

**Reply by ludwa6** [reply]
I'd like to reinforce your item #4 above, @NicolasPetton -i.e.: 
> Add documentation about implementing specific workflows (**_GTD setup_**, etc.)

(emphasis on the _GTD setup_ UseCase mine).  **Moreover**: if there could be a video tutorial demonstrated workflow related to doing a morning "mind-sweep" and perhaps a week-end review, by somebody who is actually using the tool in a very agile way, that would be both instructive *and* a great inspiration for those of us still struggling to be productive with this tool.

**Reply by NicolasPetton** [reply]
> for those of us still struggling to be productive with this tool

Are you struggling to be productive with Projectify? If so, would mind explaining what you feel is lacking/confusing?

**Reply by ludwa6** [reply]
Sure, @NicolasPetton; for example:  [here's one](https://github.com/NicolasPetton/Projectify/discussions/22) that's quite specific. 

More generally: for this to serve as my "trusted system" (in GTD parlance), it needs a few things that i'm still thinking about how to address, perhaps most importantly:

* Contexts: a fundamental concept in GTD, one should only have to deal with things addressable in one's current context... And while Projectify works great in my desktop Context, i do have other Contexts in my life.  That brings me to...
* Mobile application:  I've got Quine2 on my iPhone (2-way synced with my TiiddlyDesktop on Mac), which serves pretty well in most of those other Contexts in my life... Except it doesn't support the Alt-C keycombo for chucking things into my Inbox, AFAICT, and UI on that platform doesn't lend itself so well to working w/ Projectify.

NB: Those are just a few of the issues i'm struggling with -not to say they are _Issues_ for Projectify project to address, necessarily; probably more a function of too much tech for a non-tech user to find his way around all at once.

To be clear: i still can't believe how much utility Projectify plugin adds to TiddlyWiki, and i'm very much a fan of the project!

**Reply by NicolasPetton** [reply]
> Contexts: a fundamental concept in GTD, one should only have to deal with things addressable in one's current context... And while Projectify works great in my desktop Context, i do have other Contexts in my life. That brings me to...

There's no support for contexts yet, but that's on the roadmap :)

> Mobile application: I've got Quine2 on my iPhone (2-way synced with my TiiddlyDesktop on Mac), which serves pretty well in most of those other Contexts in my life... Except it doesn't support the Alt-C keycombo for chucking things into my Inbox, AFAICT, and UI on that platform doesn't lend itself so well to working w/ Projectify.

Have you tried the [Notebook](https://nicolas.petton.fr/tw/notebook.html) theme? Is has a dedicated layout for mobile devices. 
Or is it something else than the layout that is bothering you?

**Reply by ludwa6** [reply]
I just installed Notebook theme (thanks for that, @NicolasPetton ) and gave it a whirl in Quine2 on my iPhone; works great there!  
If only i could set Quine2 and TiddlyDesktop to use two different themes, that would be great... But unfortunately, they are sharing a single index.html file (synced via iCloud), so i would have to go through the hassle of changing theme & color palette, every time i switch devices.  Guess i'll stick w/ plain vanilla theme & color palette, which works great on TiddlyDesktop, and not too bad on mobile (am [talking to the Quine2 users group](https://groups.google.com/u/1/g/quine-app/c/tj4SQds7Yjk) about how to make that experience better).

**Reply by NicolasPetton** [reply]
@ludwa6 Projectify has categories for projects now. It might not be enough for GTD contexts, but labels on todos will be landing soon.

**Reply by jcholsap** [reply]
> 5. Give an option to assign a specific time (like 10 am) or specific period (morning or evening or night) of the due date selected for that todos.

Would it be much work to extend the date field to include hours and minutes?

**Comment by lfilho** [comment]
Nicola, let me know if you want me to open separate issue or discussions for each one of the below, had to do that if you'd prefer it. The features below are more "enterprise"-y, with teams and big projects in mind...

1. **Effort / Effort Remaining.**
2. **Task Dependency.** A way to tell task A depends on task B
3. With 1 and 2, we could then **Forecast completion date** based on the effort
4. With 1-3, we could plot a **Gantt chart** and know when project could end
5. **Project nesting**. This is so we could see the health of a big, year-long project from a high level ("60 tasks remaining, 200 days remaining") -- good for reporting, as well to drill down on it and see the healthy and track its individual components/milestones. In other words, what I'm imagining it's a table (project) that you can import/transclude other tables (sub-projects) into it

**Reply by NicolasPetton** [reply]
Thanks for the feedback @lfilho!

I do have time tracking (both estimates and actuals) in the roadmap for Projectify, as well as support for collaboration (assigning tasks).
Task dependency is not on my roadmap currently, simply because I have no need for it, but if enough users request it, I might add support for that in the future.

Regarding project nesting, today there's support for adding sub-tasks to a task, which should be enough, except that it won't help with reporting or giving a high-level overview of a project.

I've been toying with the idea of having a dedicated view for project categories though. It would list all projects in a category, show simple statistics, things like that. Would that fit your need?

**Reply by NicolasPetton** [reply]
Regarding Gantt charts, I consider it out of the scope of Projectify, but I would be very glad if someone was willing to contribute that as a plugin that would work on top of it. I might very well build that myself one day, but I can't promise anything, as there are *many* things I'd consider more important and urgent to add to Projectify :-)

**Comment by Bing-NotChandler** [comment]
Similarly to @arunnbabu81 's suggestion, having timestamps be created when a new 'to do' or 'project' is made, in a similar fashion to the Kookma Todolist plugin. 
Please forgive me if I am mistaken and this is already a feature of projectify, or something that can be easily done by one's self. I am quite new to Tiddlywiki.
Love this plugin btw!

**Comment by Telumire** [comment]
First of all, I want to thank you, your plugin is amazing! I've been trying to optimize my task management with TiddlyWiki for several months now, and your solution meets almost all my needs. I am very grateful for the work you bring to the community. The design is very nice as well.

I have a few suggestions:

1. Rather than having the "add subtasks" and "convert to a project" buttons into a drop-down menu, displaying the icons directly would be better: currently there are only 4 icons so an overflow menu is a bit superfluous (in desktop mode) IMO.

2. A button in the View Toolbar to convert an existing tiddler into a task without needing to open it and add a tag manually would be ideal.

3. Probably too complicated to implement, but exporting the tasks in .ical format so they can be displayed in an electronic calendar would be amazing. I use google calendar a lot to manage my tasks, it gives a very visual idea of how much space the tasks take up during the day. This feature would allow Projectify and Google Calendar to communicate (or any other similar calendar service).

4. On PC, with firefox, when the browser window is exactly 960px wide, the hamburger button disappears. It's not a big problem as you just have to resize the window a bit but it should be relatively easy to fix.

5. I understand the logic of making the menu appear at the very bottom of the screen when the screen is less than 960px wide, it's suitable for mobile browsing. However, on PC the mouse cursor usually stays around the top of the screen, especially in TiddlyWiki because of the way the story fills up from the top to the bottom. Furthermore, all the other toolbars are on the top of the page. So in order to be more consistent and avoid unnecessary mouse movements, I think it would be better to leave it at the top of the screen.

EDIT: Since you want to implement a Kanban, maybe this example could help a bit? It's quite crude but it does the job : 

https://tiddlytask.github.io/#:Kanban%20%5B%5BStyle%20Kanban%5D%5D%20%5B%5BStyle%20Tiddlytask%20Color%20Palette%5D%5D%20%5B%5BStyle%20Tiddlytask%20Symbols%5D%5D
